code/game/objects/items/mop.dm

```diff
@@ -29,24 +29,25 @@
 	GLOB.janitor_devices -= src
 	return ..()
 
-/obj/item/mop/proc/clean(turf/A, mob/living/cleaner)
-	if(reagents.has_chemical_flag(REAGENT_CLEANS, 1))
-		// If there's a cleaner with a mind, let's gain some experience!
-		if(cleaner?.mind)
-			var/total_experience_gain = 0
-			for(var/obj/effect/decal/cleanable/cleanable_decal in A)
-				//it is intentional that the mop rounds xp but soap does not, USE THE SACRED TOOL
-				total_experience_gain += max(round(cleanable_decal.beauty / CLEAN_SKILL_BEAUTY_ADJUSTMENT, 1), 0)
-			cleaner.mind.adjust_experience(/datum/skill/cleaning, total_experience_gain)
-		A.wash(CLEAN_SCRUB)
-
-	reagents.expose(A, TOUCH, 10) //Needed for proper floor wetting.
+/obj/item/mop/ComponentInitialize()
```

Comment:
Can you use Initialize() instead of ComponentInitialize() ? It's deprecated and we would like to remove it in the future.